specs/clipboard-api/clipboard.idl

```diff
@@ -0,0 +1,445 @@
+  delegate DataProviderHandler;
+  runtimeclass Clipboard;
+  runtimeclass ClipboardContentOptions;
+  runtimeclass ClipboardHistoryChangedEventArgs;
+  runtimeclass ClipboardHistoryItem;
+  runtimeclass ClipboardHistoryItemsResult;
+  runtimeclass DataPackage;
+  runtimeclass DataPackagePropertySet;
+  runtimeclass DataPackagePropertySetView;
+  runtimeclass DataPackageView;
+  runtimeclass DataProviderDeferral;
+  runtimeclass DataProviderRequest;
+  runtimeclass HtmlFormatHelper;
+  runtimeclass OperationCompletedEventArgs;
+  runtimeclass StandardDataFormats;
+}
+
+// Generic instantiations
+namespace Microsoft.ProjectReunion.ApplicationModel
+{
+  declare
+  {
+    interface Windows.Foundation.Collections.IIterable<ClipboardHistoryItem*>;
+    interface Windows.Foundation.Collections.IIterator<ClipboardHistoryItem*>;
+    interface Windows.Foundation.Collections.IVectorView<ClipboardHistoryItem*>;
+    interface Windows.Foundation.EventHandler<ClipboardHistoryChangedEventArgs*>;
+    interface Windows.Foundation.IAsyncOperation<ClipboardHistoryItemsResult*>;
+    interface Windows.Foundation.IAsyncOperation<DataPackage*>;
+    interface Windows.Foundation.IAsyncOperation<DataPackageOperation>;
+    interface Windows.Foundation.TypedEventHandler<DataPackage*, IInspectable*>;
+    interface Windows.Foundation.TypedEventHandler<DataPackage*, OperationCompletedEventArgs*>;
+  }
+}
+
+// Type definitions
+namespace Microsoft.ProjectReunion.ApplicationModel
+{
+  // FIXME: Figure out how to declare these enums without causing definition conflicts
+  // between the enum constants and the same-named ones in
+  // Windows.ApplicationModel.DataTransfer
+
+  [contract(ClipboardContract, 1.0)]
+  enum ClipboardHistoryItemsResultStatus
+  {
+    aaaSuccess                  = 0,
+    aaaAccessDenied             = 1,
+    aaaClipboardHistoryDisabled = 2,
+  };
+
+  [contract(ClipboardContract, 1.0)]
+  [flags]
+  enum DataPackageOperation
+  {
+    // TODO: combine Copy and Move?
+    aaaNone = 0x0,
+    aaaCopy = 0x1,
+    aaaMove = 0x2,
+    aaaLink = 0x4,
+  };
+
+  [contract(ClipboardContract, 1.0)]
+  enum SetHistoryItemAsContentStatus
+  {
+    aaaSuccess      = 0,
+    aaaAccessDenied = 1,
+    aaaItemDeleted  = 2,
+  };
+
+  /// <summary>
+  /// Prototype for functions that will be called to render a delay-rendered
+  /// data format in a <see cref="DataPackage"/>.
+  /// </summary>
+  [contract(ClipboardContract, 1.0)]
+  [uuid(6d9e7bfb-3280-4219-be29-08145f30f4e5)]
+  delegate void DataProviderHandler(DataProviderRequest request);
+
+  /// <summary>
+  /// Top-level API for reading, writing, and monitoring the system clipboard
+  /// and clipboard history, if available. Clipboard history also
+  /// includes clipboard items roamed from your other devices through the cloud.
+  /// </summary>
+  [contract(ClipboardContract, 1.0)]
+  [static_name("IClipboardStatics", 6cb53030-4da0-43a1-95ab-52f87cf59c3a)]

```

Comment:
>[static_name("IClipboardStatics", 6cb53030-4da0-43a1-95ab-52f87cf59c3a)] [](start = 2, length = 72)

remove all custom uuids, cppwinrt will generate it

static_name, interface_name

Reply:
normally I'm not a fan of letting midlrt.exe generate the GUIDs, it makes it impossible to search for them in the original source code (.idl) itself and you have to generate the .h first.

but it's not a big deal, so I removed them.